app/firewall.js

```diff
@@ -8,22 +8,37 @@ const {isInternalUrl} = require('../js/lib/urlutil')
 
 module.exports.resourceName = 'firewall'
 
-const onHeadersReceived = (details) => {
-  const result = { resourceName: module.exports.resourceName }
-  const mainFrameUrl = Filtering.getMainFrameUrl(details)
-  const isIPInternal = ip.isPrivate(details.ip)
-  const isUrlInternal = isInternalUrl(details.url)
+/**
+ * Export this so it can be unit tested.
+ * @param {string} url
+ * @param {string} mainFrameUrl
+ * @param {boolean} isPrivate
```

Comment:
nit: missing the `ipAddr` param - might be useful to have a quick description for params (ex: how is url different than mainFrameUrl)